### ISSUE TITLE
Error: Uncaught (in promise): false errors

### ISSUE THREAD

**Issue by basvdijk** [ionic-team/ionic#11341]
**Ionic version:**  (check one with "x")
[ ] **1.x**
[ ] **2.x**
[X] **3.x**

**I'm submitting a ...**  (check one with "x")
[X] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://forum.ionicframework.com/ or http://ionicworldwide.herokuapp.com/

**Current behavior:**

In general I see a lot errors in Sentry of which I think are all related to toast and popups. Which might also be related to the NavController. I cannot reproduce these since they occur on my customers app installations.

```
Error: Uncaught (in promise): false
  at g (/build/polyfills.js:3:7133)
  at l (/build/polyfills.js:3:6251)
  at l (/build/polyfills.js:3:5937)
  at None (/build/polyfills.js:3:6765)
  at t.invokeTask (/build/polyfills.js:3:15213)
```

```
Error: Uncaught (in promise): false
  at g (/android_asset/www/build/polyfills.js:3:7133)
  at l (/android_asset/www/build/polyfills.js:3:6251)
  at None (/android_asset/www/build/polyfills.js:3:5607)
  at Object.e.reject (/android_asset/www/build/main.js:25:10925)
  at t._postViewInit (/android_asset/www/build/main.js:25:13148)
```

**Expected behavior:**

**Ionic info:** (run `ionic info` from a terminal/cmd prompt and paste output below):

```
Cordova CLI: 6.5.0
Ionic Framework Version: 3.0.1
Ionic CLI Version: 2.2.2
Ionic App Lib Version: 2.2.1
Ionic App Scripts Version: 1.3.3
ios-deploy version: Not installed
ios-sim version: 5.0.13
OS: OS X El Capitan
Node Version: v7.2.0
Xcode version: Xcode 8.2.1 Build version 8C1002
```

**Comment by manucorporat** [comment]
@basvdijk  Are you using ionViewCanLeave or ionViewCanEnter?

**Comment by basvdijk** [comment]
@manucorporat No nothing of these two. I had this problem before, first I had:
```
toast.present();
this.nav.pop();
```
then later I tried

```
this.nav.pop().then(() => toast.present());
```

**Comment by manucorporat** [comment]
So you don't use `ionViewCanLeave/ionViewCanEnter` in your app at all?

why do you think it is related with Toast?

**Comment by manucorporat** [comment]
```
this.nav.pop().then(() => toast.present());
```

works but:
```
toast.present();
this.nav.pop();
```
does not?

**Comment by basvdijk** [comment]
No they both work, I thought the problem was the present() was not finished while I already was navigating to another page.

I use Alert and Toast through my app. I've got the `Uncaught (in promise): false` sometimes while developing. This was when I wanted to show a toast or an alert. Therefore I thought it is related to one of these when I saw the same errors in Sentry.

Also since the error doesn't occur all the time and imho quite random I guess it could be a race condition. This race condition could make sense for presenting a toast and navigating away from the page.
